pulp_container/app/tasks/sync_stages.py

```diff
@@ -629,19 +640,19 @@ def _post_save(self, batch):
                 manifest_lists.append(dc.content)
                 for listed_manifest in dc.extra_data["listed_manifests"]:
                     manifest_dc = listed_manifest["manifest_dc"]
-                    platform = listed_manifest["platform"]
-                    manifest_list_manifests.append(
-                        ManifestListManifest(
-                            manifest_list=manifest_dc.content,
-                            image_manifest=dc.content,
-                            architecture=platform["architecture"],
-                            os=platform["os"],
-                            features=platform.get("features"),
-                            variant=platform.get("variant"),
-                            os_version=platform.get("os.version"),
-                            os_features=platform.get("os.features"),
-                        )
+                    manifest_list_manifest = ManifestListManifest(
+                        manifest_list=manifest_dc.content,
+                        image_manifest=dc.content,
                     )
+                    if platform := listed_manifest.get("platform"):
+                        manifest_list_manifest.architecture = platform["architecture"]
+                        manifest_list_manifest.os = platform["os"]
+                        manifest_list_manifest.features = platform.get("features")
+                        manifest_list_manifest.variant = platform.get("variant")
+                        manifest_list_manifest.os_version = platform.get("os.version")
+                        manifest_list_manifest.os_features = platform.get("os.features")
```

Comment:
This is at least the third time i'm seeing this pattern now.

Maybe a dataclass together with `asdict` can help here? (Maybe not in this PR.)
https://docs.python.org/3/library/dataclasses.html

Reply:
We discussed this and agreed on opening a task issue to investigate this later.

Reply:
I do not understand why this cannot be addressed in this PR. Also, notice how in some cases you are initializing `os_version` as `p.get("os.version")` and in some cases as `p.get("os.version", "")`. Not to mention that you even do `(p.get("os.version", ""),)`. I am not a fan of such discrepancies. Please, resolve this.

Reply:
I tried to use the `@dataclass` decorator, but I had trouble to make it work (probably because of the `image_manifest` and `manifest_list` foreignkey fields).